chainerx_cc/chainerx/python/array.cc

```diff
@@ -240,25 +240,38 @@ void InitChainerxArrayConversion(pybind11::module& m, py::class_<ArrayBody, Arra
 
 void InitChainerxArrayManipulation(py::class_<ArrayBody, ArrayBodyPtr>& c) {
     c.def("take",
-          [](const ArrayBodyPtr& self, py::handle indices, const absl::optional<int8_t>& axis) {
+          [](const ArrayBodyPtr& self, py::handle indices, const absl::optional<int8_t>& axis, absl::optional<std::string>& mode) {
               if (!axis.has_value()) {
                   throw NotImplementedError{"axis=None is not yet supported for chainerx.ndarray.take."};
               }
+              IndexBoundsMode tmode{};
+              if (!mode.has_value()) {
+                  tmode = IndexBoundsMode::kDefault;
+              } else if (mode.value() == "raise") {
+                  tmode = IndexBoundsMode::kRaise;
+              } else if (mode.value() == "wrap") {
+                  tmode = IndexBoundsMode::kWrap;
+              } else if (mode.value() == "clip") {
+                  tmode = IndexBoundsMode::kClip;
+              } else {
+                  throw py::value_error{"mode must be 'raise', 'wrap', or 'clip'"};
+              }
```

Comment:
[`.value()` does an existence check](https://github.com/abseil/abseil-cpp/blob/master/absl/types/optional.h#L463-L490), let's just check it once in favor of performance, maybe assign it to a local.

Reply:
Fixed